src/ModalForm.js

```diff
@@ -0,0 +1,109 @@
+import React, {PropTypes} from 'react';
+import {
+  Button,
+  Modal,
+  FormGroup,
+  ControlLabel,
+  FormControl,
+  HelpBlock
+} from 'react-bootstrap';
+
+function FieldGroup({ id, label, help, ...props }) {
+  return (
+    <FormGroup controlId={id}>
+      <ControlLabel>{label}</ControlLabel>
+      <FormControl {...props} />
+      {help && <HelpBlock>{help}</HelpBlock>}
+    </FormGroup>
+  );
+}
+
+const ModalForm = React.createClass({
+
+  getInitialState() {
+    return {
+      showModal: false,
+    };
+  },
+
+  close() {
+    this.setState({ showModal: false });
+  },
+
+  open() {
+    this.setState({ showModal: true });
+  },
+
+  handleSubmit(event) {
+    event.preventDefault();
+
+    const {title, description, buttonText, url} = this.state;
+    this.props.onSubmit({title, description, buttonText, url});
+
+    this.setState({ showModal: false });
+  },
+
+  render() {
+    return (
+      <div>
+        <Button
+          bsStyle="info"
+          onClick={this.open}
+        >
+          Add Content
+        </Button>
+
+        <Modal show={this.state.showModal} onHide={this.close}>
+          <Modal.Header closeButton />
+          <Modal.Body>
+            <form onSubmit={this.handleSubmit}>
+              <FieldGroup
+                id="formControlsText"
+                type="text"
+                label="Title"
+                placeholder="Enter title"
+                value={this.title}
```

Comment:
Finally, we need to modify our field group to call the onInputChange function on every change, additionally we need to add a "name" property to the fieldgroup so onInputChange can identify the field. 

It'll look like this: 

``` 
  <FieldGroup
                id="formControlsText"
                type="text"
                label="Title"
                name="title"
                placeholder="Enter title"
                value={this.state.title}
                onChange={(evt) => this.onInputChange(evt)}
              />
```